pkg/controllers/management/clustertemplate/clustertemplate_rbac.go

```diff
@@ -0,0 +1,128 @@
+	ctLabel      = "io.cattle.field/clusterTemplateId"
+)
+
+type clusterTemplateController struct {
+	clusterTemplates              v3.ClusterTemplateInterface
+	clusterTemplateLister         v3.ClusterTemplateLister
+	clusterTemplateRevisionLister v3.ClusterTemplateRevisionLister
+	managementContext             *config.ManagementContext
+}
+
+type clusterTemplateRevisionController struct {
+	clusterTemplateRevisions      v3.ClusterTemplateRevisionInterface
+	clusterTemplateRevisionLister v3.ClusterTemplateRevisionLister
+	clusterTemplateLister         v3.ClusterTemplateLister
+	managementContext             *config.ManagementContext
+}
+
+func Register(ctx context.Context, mgmt *config.ManagementContext) {
+	ct := clusterTemplateController{
+		managementContext:             mgmt,
+		clusterTemplates:              mgmt.Management.ClusterTemplates(""),
+		clusterTemplateLister:         mgmt.Management.ClusterTemplates("").Controller().Lister(),
+		clusterTemplateRevisionLister: mgmt.Management.ClusterTemplateRevisions("").Controller().Lister(),
+	}
+	ct.clusterTemplates.AddHandler(ctx, "cluster-template-rbac-controller", ct.sync)
+
+	ctr := clusterTemplateRevisionController{
+		clusterTemplateRevisions:      mgmt.Management.ClusterTemplateRevisions(""),
+		clusterTemplateRevisionLister: mgmt.Management.ClusterTemplateRevisions("").Controller().Lister(),
+		clusterTemplateLister:         mgmt.Management.ClusterTemplates("").Controller().Lister(),
+	}
+	ctr.clusterTemplateRevisions.AddHandler(ctx, "cluster-template-rev-rbac-controller", ctr.sync)
+}
+
+func (ct *clusterTemplateController) sync(key string, clusterTemplate *v3.ClusterTemplate) (runtime.Object, error) {
+	if clusterTemplate == nil || clusterTemplate.DeletionTimestamp != nil {
+		return nil, nil
+	}
+	metaAccessor, err := meta.Accessor(clusterTemplate)
+	if err != nil {
+		return clusterTemplate, err
+	}
+	creatorID, ok := metaAccessor.GetAnnotations()[globalnamespacerbac.CreatorIDAnn]
+	if !ok {
+		return clusterTemplate, fmt.Errorf("clusterTemplate %v has no creatorId annotation", metaAccessor.GetName())
+	}
+
+	if err := globalnamespacerbac.CreateRoleAndRoleBinding(globalnamespacerbac.ClusterTemplateResource, clusterTemplate.Name,
+		globalnamespacerbac.RancherManagementAPIVersion, creatorID, []string{globalnamespacerbac.RancherManagementAPIVersion},
+		clusterTemplate.UID,
+		clusterTemplate.Spec.Members, ct.managementContext); err != nil {
+		return nil, err
+	}
+
+	//see if any revision exists and call CreateRoleAndRoleBinding for it so that if this is an update request adding/removing members,
+	//it reflects on the permissions for revisions too
+	revisions, err := ct.clusterTemplateRevisionLister.List(namespace.GlobalNamespace, labels.SelectorFromSet(map[string]string{ctLabel: clusterTemplate.Name}))
+	if err != nil && !apierrors.IsNotFound(err) {
+		return clusterTemplate, err
+	}
+	for _, rev := range revisions {
+		metaAccessor, err := meta.Accessor(rev)
```

Comment:
can we just enqueue the revision? And let the `func (ctr *clusterTemplateRevisionController) sync(key string...` to set the rbac rules on per revision basis